covid_moonshot_ml/nn/loss/GaussianNLLLoss.py

```diff
@@ -0,0 +1,61 @@
+import torch
+from torch.nn import GaussianNLLLoss as TorchGaussianNLLLoss
+
+
+class GaussianNLLLoss(TorchGaussianNLLLoss):
+    def __init__(self, include_semiquant=True, fill_value=None):
+        """
+        Class for calculating Gaussian NLL loss, with various options.
+
+        Parameters
+        ----------
+        include_semiquant : bool, default=True
+            Whether to include semi-quantitative samples in the loss
+        fill_value : float, optional
+            If provided, use this value as the uncertainty for all
+            semiquant predictions
+        """
+        ## No reduction so we can apply masking if desired
+        super(GaussianNLLLoss, self).__init__(reduction="none")
+
+        self.include_semiquant = include_semiquant
+        self.fill_value = fill_value
+
+    def forward(self, input, target, in_range, uncertainty):
+        """
+        Loss calculation
+
+        Parameters
+        ----------
+        input : torch.Tensor
+            Model prediction
+        target : torch.Tensor
+            Prediction target
+        in_range : torch.Tensor
+            `target`'s presence in the dynamic range of the assay. Give a value
+            of < 0 for `target` below lower bound, > 0 for `target` above upper
+            bound, and 0 or None for inside range
+        uncertainty : torch.Tensor
+            Uncertainty in `target` measurements
+
+        Returns
+        -------
+        """
+        ## Fill in semiquant values
+        if self.include_semiquant and self.fill_value is not None:
+            idx = [r != 0 for r in in_range]
+            ## Clone to avoid modifying the original uncertainty measurements
+            uncertainty = uncertainty.clone()
```

Comment:
I might be misunderstanding, but would it not be safer to use e.g. `uncertainty_clone` and continue with that? Not sure how the original variable is handled/whether it can be overwritten by the current design?

Reply:
I don't think it's necessary, but I've added this in [c720b64](https://github.com/choderalab/covid-moonshot-ml/pull/33/commits/c720b64c30399f5d831b6f4e70d2befd2b405601) just to be safe